src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -415,9 +416,29 @@ private unsafe Span<byte> GetUnknownMethod(byte* data, int length, out int metho
             return new Span<byte>(data, methodLength);
         }
 
+        private unsafe bool IsTlsHandshake(byte* data, int length)
+        {
+            const byte SslRecordTypeHandshake = (byte)0x16;
+
+            //make sure we can check at least for the existence of a TLS handshake - we check the first three bytes
```

Comment:
Style:
```suggestion
            // Make sure we can check at least for the existence of a TLS handshake - we check the first byte
```

Reply:
@Tratcher 

The next commit is ready. Looking forward for your review.